src/swap/components/SwapSettingsSlippageLayout.tsx

```diff
@@ -0,0 +1,55 @@
+import { Children, useMemo } from 'react';
+import type { ReactNode } from 'react';
+import { findComponent } from '../../internal/utils/findComponent';
+import { cn } from '../../styles/theme';
+import { SwapSettingsSlippageDescription } from './SwapSettingsSlippageDescription';
+import { SwapSettingsSlippageInput } from './SwapSettingsSlippageInput';
+import { SwapSettingsSlippageLabel } from './SwapSettingsSlippageLabel';
+import { SwapSettingsSlippageToggle } from './SwapSettingsSlippageToggle';
+
+type SwapSettingsSlippageLayoutReact = {
+  children: ReactNode;
+  className?: string; // Optional className override for top div element.
+};
+
+export function SwapSettingsSlippageLayout({
+  children,
+  className,
+}: SwapSettingsSlippageLayoutReact) {
+  const { label, description, toggle, input } = useMemo(() => {
+    const childrenArray = Children.toArray(children);
+    return {
+      label: childrenArray.find(findComponent(SwapSettingsSlippageLabel)),
+      description: childrenArray.find(
+        findComponent(SwapSettingsSlippageDescription),
+      ),
+      toggle: childrenArray.find(findComponent(SwapSettingsSlippageToggle)),
+      input: childrenArray.find(findComponent(SwapSettingsSlippageInput)),
+    };
+  }, [children]);
+
+  return (
+    <div
+      className={cn(
+        'right-0 z-10 w-[21.75rem] rounded-lg border border-gray-300 bg-gray-50 p-4 shadow-lg dark:border-gray-700 dark:bg-gray-950',
+        className,
+      )}
+      data-testid="ockSwapSettingsLayout_container"
+    >
+      {label && (
+        <h3 className="mb-2 font-semibold text-base text-gray-950 leading-normal dark:text-gray-50">
+          {label}
+        </h3>
+      )}
+      {description && (
+        <p className="mb-4 font-normal font-sans text-gray-600 text-xs leading-4 dark:text-gray-400">
+          {description}
+        </p>
+      )}
+      <div className="flex items-center justify-between gap-2">
+        <div className="flex-grow">{toggle}</div>
+        {input}
```

Comment:
what kind of input is this?

Reply:
Text input. This is where the user puts in their preferred slippage percent